app/src/main/java/mainburg/planetenweg/LocationMarker.java

```diff
@@ -0,0 +1,146 @@
+package mainburg.planetenweg;
+
+import android.*;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.widget.Toast;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * The LocationMarker places a marker on your current location and
+ * updates it after a certain interval.
+ * Created by Markus on 04.10.2017.
+ */
+public class LocationMarker implements LocationListener {
+
+    private GoogleMap map;
+    private final Activity activity;
+    private final LocationManager gps;
+    private Marker position;
+    static final int GPS_PERMISSION_REQUEST_CODE = 1227;
+
+    private boolean active;
+
+    /**
+     * This variable indicates whether or not a location has been founb.
+     * It is needed to notify the user when his location could not have been determined.
+     */
+    private boolean locationFound;
+
+    public LocationMarker(Activity activity) {
+        this.map = map;
+        this.activity = activity;
+        active = false;
+        gps = (LocationManager) activity.getSystemService(Context.LOCATION_SERVICE);
+    }
+
+    /**
+     * Enables the Location Marker
+     * @param map The GoogleMap on which to place the marker
+     */
+    public void enable(GoogleMap map) {
```

Comment:
Is this the implementation of the interface :top: ?

Reply:
Nope.